### PR TITLE
🦢 Swap sign up step

### PR DESCRIPTION
**Pull request by WRadoslaw**
#4289

### CODE REVIEW

File: packages/atlas/src/components/_auth/AuthModals/AuthModals.tsx [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
-  switch (authModalOpenName) {
-    case 'logIn':
-      return <LogInModal />
-    case 'externalLogIn':
-      return <ExternalSignInModal />
-    case 'signUp':
-      return <SignUpModal />
-    default:
-      return null
+  const debouncedAuthModalOpenName = useDebounceValue(authModalOpenName, 2_000)
```
Comment (drillprop) [inline packages/atlas/src/components/_auth/AuthModals/AuthModals.tsx]: I would remove that, it doesn't work well, IMO it's worse than before

File: packages/atlas/src/components/_auth/ExternalSignInModal/ExternalSignInSteps/ExternalSignInModalEmailStep.tsx [hunk 2]
```diff
@@ -34,30 +38,61 @@ export const ExternalSignInModalEmailStep: FC<SignInModalEmailStepProps> = ({
   const { handleSubmit, setError, formState, register } = useFormContext<{ email: string }>()
   const handleConfirm = useCallback(async () => {
     const account = await joystream?.selectedAccountId
-    if (!joystream?.signMessage || !account || !memberId) return
+    if (!joystream?.signMessage || !account || !memberId || !formState.isValid) return
```
Comment (drillprop) [inline packages/atlas/src/components/_auth/ExternalSignInModal/ExternalSignInSteps/ExternalSignInModalEmailStep.tsx]: As far as I know, we don't need to add another check here. If `formState` is invalid it shouldn't run handleSubmit, this is handled by react-hook-form

File: packages/atlas/src/components/_auth/SignUpModal/SignUpSteps/SignUpEmailStep.tsx [hunk 3]
```diff
@@ -78,6 +81,12 @@ export const SignUpEmailStep: FC<SignUpEmailStepProps> = ({
     })
   }, [handleGoToNextStep, isEmailAlreadyTakenError, setPrimaryButtonProps])
 
+  useMountEffect(() => {
+    if (isEmailAlreadyTakenError) {
+      setError('email', { message: 'Email already taken' })
```
Comment (drillprop) [inline packages/atlas/src/components/_auth/SignUpModal/SignUpSteps/SignUpEmailStep.tsx]:
```suggestion
      setError('email', { message: 'Email already taken.' })
```

Comment (drillprop) [reply packages/atlas/src/components/_auth/SignUpModal/SignUpSteps/SignUpEmailStep.tsx]: For consistency, we are using dots on every validation message.

File: packages/atlas/src/hooks/useCreateMember.ts [hunk 4]
```diff
@@ -37,16 +52,22 @@ type FaucetParams = {
...
+  onSuccess?: (amountOfTokens?: number) => void
+  onError?: (step?: RegisterError) => void
+}
+
 export const useCreateMember = () => {
   const { handleLogin } = useAuth()
-  const { refetchUserMemberships } = useUser()
-  const [emailAlreadyRegisteredMemberId, setEmailAlreadyRegisteredMemberId] = useState('')
+  // const { refetchUserMemberships } = useUser()
+  // const [emailAlreadyRegisteredMemberId, setEmailAlreadyRegisteredMemberId] = useState('')
```
Comment (drillprop) [inline packages/atlas/src/hooks/useCreateMember.ts]: to remove?

File: packages/atlas/src/hooks/useCreateMember.ts [hunk 5]
```diff
@@ -199,20 +161,67 @@ export const useCreateMember = () => {
...
+      onStart?.()
+
+      await cryptoWaitReady()
+      const keypair = keyring.addFromMnemonic(data.mnemonic)
+
+      const address = keypair.address
+
+      try {
+        // const { data: memberShipData } = await refetchUserMemberships()
+        // const lastCreatedMembership = memberShipData.memberships[memberShipData.memberships.length - 1]
```
Comment (drillprop) [inline packages/atlas/src/hooks/useCreateMember.ts]: to remove as well? I know you asked me about this and we had a condition here, but can't recall why we actually needed that, so let's just remove this

**Review by drillprop — CHANGES_REQUESTED** [review]

File: packages/atlas/src/components/_auth/AuthModals/AuthModals.tsx (hunk 1 above)
Comment (WRadoslaw) [reply packages/atlas/src/components/_auth/AuthModals/AuthModals.tsx]: New approach in commit `CR fixes`

File: packages/atlas/src/components/_auth/SignUpModal/SignUpSteps/SignUpEmailStep.tsx [hunk 6]
```diff
@@ -53,9 +54,11 @@ export const SignUpEmailStep: FC<SignUpEmailStepProps> = ({
     register,
     handleSubmit,
     control,
+    setError,
     formState: { errors },
   } = useForm<EmailStepForm>({
     criteriaMode: 'all',
+    mode: 'onBlur',
```
Comment (drillprop) [inline packages/atlas/src/components/_auth/SignUpModal/SignUpSteps/SignUpEmailStep.tsx:61]: why did we change the mode here?

Comment (WRadoslaw) [reply packages/atlas/src/components/_auth/SignUpModal/SignUpSteps/SignUpEmailStep.tsx:61]: After setting the field error when email is taken, it wouldn't revalidate the field, so even if the user has changed the email, the error would still be visible

**Review by drillprop — APPROVED** [review]

**Pull request merged**